loadtest/control/simplecontroller/config.go

```diff
@@ -0,0 +1,115 @@
+package simplecontroller
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/control"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/user"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+var v = viper.New()
+
+type Config struct {
+	Rate    float32            `json:"Rate"`
+	Actions []actionDefinition `json:"Actions"`
+}
+
+type actionDefinition struct {
+	ActionId     string `josn:"ActionId"`
+	RunFrequency int    `json:"RunFrequency"`
+	WaitAfterMs  int    `json:"WaitAfterMs"`
+}
+
+func ReadConfig(configFilePath string) error {
+	v.SetConfigName("simplecontroller")
+	v.AddConfigPath(".")
+	v.AddConfigPath("./config/")
+	v.AddConfigPath("./../config/")
+	v.AddConfigPath("./../../../config/")
+	v.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	v.AutomaticEnv()
+
+	if configFilePath != "" {
+		v.SetConfigFile(configFilePath)
+	}
+
+	if err := v.ReadInConfig(); err != nil {
+		return errors.Wrap(err, "unable to read configuration file")
+	}
+
+	return nil
+}
+
+func GetConfig() (*Config, error) {
+	var cfg *Config
+
+	if err := v.Unmarshal(&cfg); err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+// IsValid checks whether a Config is valid or not.
+// Returns an error if the validation fails.
+func (c *Config) IsValid() error {
+	if _, err := parseActions(&SimpleController{}, c.Actions); err != nil {
+		return fmt.Errorf("actions are not valid: %w", err)
+	}
+	return nil
+}
+
+func parseActions(c *SimpleController, definitions []actionDefinition) ([]*UserAction, error) {
+	actions := make([]*UserAction, 0)
+	actionMap := map[string]control.UserAction{
```

Comment:
Since the `simplecontroller.json` configuration file is an external file, we don't need a proper godoc for it, do we? We just need to document it somewhere and we can use a single local variable to map strings to functions.

Reply:
I see. I thought we were serving the code itself as a documentation. Fine then, this needs to be documented somewhere. There is a docs directory we can use.

Reply:
It seems more exquisite to add it to the code, but at the end of the day you are still maintaining two vars and need to check their consistency :) I would like to hear @streamer45 as well :)

Reply:
So, each controller has a chance to either use the common action under the `control` package or write its own implementation. Common actions were made for the sole purpose of avoiding code repetition. They are not there to force controllers to use them (they are not part of an interface).

I don't think there's any harm in having a single private map (actionId to action) in the `simplecontroller` package. The important thing is that we document all the available `actionIds` and their meaning/behaviour. I think this should be done when documenting `simplecontroller.json` since we don't really need to export anything except the config.

The other approach would be to have a map of common actions under `control` and then make the controllers define their own map with just the differences essentially (either re-implementations or new added actions). Not sure if this is similar to what @agnivade proposed.

I am slightly in favor of the single private map because while there might be more repetition it will probably cause less confusion.
The set of actions a controller supports should be decided exclusively by its implementation. 
Hence I suggest we let the implementation define what that set of actions looks like and if it needs to make it somehow user-configurable (like in this case) then it should be documented accordingly.

Reply:
Sure, that's totally fine. When I saw the 2 maps, I thought they were intended, and my solution went into keeping the 2 maps. But now that I know that it was an oversight, documenting it in the json config sounds good.